waspc/packages/wasp-config/__tests__/testFixtures.ts

```diff
@@ -71,731 +145,844 @@ type PageType = (typeof PAGE_TYPES)[number];
-    name: "FullJob",
-    config: {
-      executor: "PgBoss",
-      perform: getPerform(scope),
-      entities: [getEntity("task")],
-      schedule: getSchedule(scope),
-    },
-  } satisfies FullNamedConfig<UserApi.JobConfig>;
 }
 
 export function getEmailFromField(
   scope: "minimal",
-): MinimalConfig<UserApi.EmailFromField>;
+): MinimalConfig<TsAppSpec.EmailFromField>;
 export function getEmailFromField(
   scope: "full",
-): FullConfig<UserApi.EmailFromField>;
+): FullConfig<TsAppSpec.EmailFromField>;
 export function getEmailFromField(
   scope: ConfigType,
-): Config<UserApi.EmailFromField> {
+): Config<TsAppSpec.EmailFromField>;
+export function getEmailFromField(
+  scope: ConfigType,
+): Config<TsAppSpec.EmailFromField> {
   if (scope === "minimal") {
     return {
       email: "test@domain.tld",
-    } satisfies MinimalConfig<UserApi.EmailFromField>;
+    } satisfies MinimalConfig<TsAppSpec.EmailFromField>;
+  } else {
+    return {
+      email: "test@domain.tld",
+      name: "ToDo App",
+    } satisfies FullConfig<TsAppSpec.EmailFromField>;
   }
-
-  return {
-    email: "test@domain.tld",
-    name: "ToDo App",
-  } satisfies FullConfig<UserApi.EmailFromField>;
 }
 
 export function getExtImport(
   scope: "minimal",
   type: AppSpec.ExtImportKind,
-): MinimalConfig<UserApi.ExtImport>;
+): MinimalConfig<TsAppSpec.ExtImport>;
 export function getExtImport(
   scope: "full",
   type: AppSpec.ExtImportKind,
-): FullConfig<UserApi.ExtImport>;
+): FullConfig<TsAppSpec.ExtImport>;
 export function getExtImport(
   scope: ConfigType,
   type: AppSpec.ExtImportKind,
-): Config<UserApi.ExtImport> {
+): Config<TsAppSpec.ExtImport>;
+export function getExtImport(
+  scope: ConfigType,
+  type: AppSpec.ExtImportKind,
+): Config<TsAppSpec.ExtImport> {
   if (type === "default") {
     if (scope === "minimal") {
       return {
         from: "@src/external",
         importDefault: "defaultExport",
-      } satisfies MinimalConfig<UserApi.ExtImport>;
+      } satisfies MinimalConfig<TsAppSpec.ExtImport>;
+    } else {
+      return {
+        from: "@src/external",
+        importDefault: "defaultExport",
+      } satisfies FullConfig<TsAppSpec.ExtImport>;
     }
-
-    return {
-      from: "@src/external",
-      importDefault: "defaultExport",
-    } satisfies FullConfig<UserApi.ExtImport>;
-  }
-
-  if (type === "named") {
+  } else if (type === "named") {
     if (scope === "minimal") {
       return {
         from: "@src/external",
         import: "namedExport",
-      } satisfies MinimalConfig<UserApi.ExtImport>;
+      } satisfies MinimalConfig<TsAppSpec.ExtImport>;
+    } else {
+      return {
+        from: "@src/external",
+        import: "namedExport",
+      } satisfies FullConfig<TsAppSpec.ExtImport>;
     }
-
-    return {
-      from: "@src/external",
-      import: "namedExport",
-    } satisfies FullConfig<UserApi.ExtImport>;
+  } else {
+    throw new Error(`Unhandled scope or type: scope=${scope}, type=${type}`);
```

Comment:
Do do we need this? Does TS enforce the correct type, or do the values come from runtime?

Reply:
Ok, fair enough. If you're wondering why, see page 15 of Effective TS: Type operations cannot effect runtime values.

Here's how we can make it clearer
```ts
// ...
   } else {
    return assertUnreachable(scope);
  }
// ...

function assertUnreachable(value: never): never {
  throw new Error(`Unhandled case: ${value}`);
}
```
See Effective TS, 2nd edition, item 59 for more details.

Finally, I'd probably change this too a switch. Personal preference, but not important.